### PR TITLE
Update Kotlin tour: Collections

### PR DESCRIPTION
**Pull request by sarahhaggarty**
To make the Collections chapter of the Kotlin tour more beginner-friendly, more explanation has been given for the key characteristics of lists, sets, and maps.

In addition, the examples given in the Maps section have been updated to make them more relatable to the real world.

### CODE REVIEW

File: docs/topics/tour/kotlin-tour-collections.md [hunk 1]
```diff
@@ -230,14 +234,23 @@ fun main() {
 
 ## Map
 
+Maps store items as key-value pairs. You access the value by referencing the key. You can imagine a map like a food menu.
+You can find the price (value), by finding the food (key) you want to eat. Maps are useful if you want to look up a value
+without using a numbered index, like in a list.
```
Comment (bravit) [inline docs/topics/tour/kotlin-tour-collections.md:239]: Great explanation!

File: docs/topics/tour/kotlin-tour-collections.md [hunk 2]
```diff
@@ -230,14 +234,23 @@ fun main() {
...
+
 To create a read-only map ([`Map`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-map/)), use the 
 [`mapOf()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-of.html) function.
 
 To create a mutable map ([`MutableMap`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-mutable-map/)),
 use the [`mutableMapOf()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/mutable-map-of.html) function.
 
 When creating maps, Kotlin can infer the type of items stored. To declare the type explicitly, add the types
-of the keys and values within angled brackets `<>` after the map declaration.
+of the keys and values within angled brackets `<>` after the map declaration. For example: `MutableMap<Int, Int>`
```
Comment (bravit) [inline docs/topics/tour/kotlin-tour-collections.md:253]: Just as in what follows, I'd suggest using `MutableMap<String, Int>` as an example here too and also adding something like "where `String` is a type of keys, and `Int` is a type of values".

File: docs/topics/tour/kotlin-tour-collections.md [hunk 3]
```diff
@@ -246,23 +259,23 @@ key and its related value:
 fun main() {
 //sampleStart
     // Read-only map
-    val readOnlyAccountBalances = mapOf(1 to 100, 2 to 100, 3 to 100)
-    println(readOnlyAccountBalances)
-    // {1=100, 2=100, 3=100}
-    
+    val readOnlyJuiceMenu = mapOf("apple" to 100, "kiwi" to 190, "orange" to 100)
+    println(readOnlyJuiceMenu)
+    // {apple=100, kiwi=190, orange=100}
```
Comment (bravit) [inline docs/topics/tour/kotlin-tour-collections.md:264]: These examples are much better indeed! Great improvement.

**Review by bravit — COMMENTED** [review]
I like what you are doing here, thanks! Besides the minor suggestion, the changes look great.

File: docs/topics/tour/kotlin-tour-collections.md [hunk 4]
```diff
@@ -230,14 +234,24 @@ fun main() {
 
 ## Map
 
+Maps store items as key-value pairs. You access the value by referencing the key. You can imagine a map like a food menu.
+You can find the price (value), by finding the food (key) you want to eat. Maps are useful if you want to look up a value
+without using a numbered index, like in a list.
+
+> * Every key in a map must be unique so that Kotlin can understand which value you want when you try to access it.
```
Comment (koshachy) [inline docs/topics/tour/kotlin-tour-collections.md:241]: I'm a bit confused with **try** here.
"Kotlin can understand which value you want to get" or "which value you want to get when you access it"?

What do you think?

**Review by koshachy — APPROVED** [review]
Thanks for this update.
See the minor comments.

File: docs/topics/tour/kotlin-tour-collections.md (hunk 4 above)
Comment (sarahhaggarty) [reply docs/topics/tour/kotlin-tour-collections.md:241]: I like your first suggestion best 👍

**Pull request merged**